devops/scripts/register_bundle_with_api.sh

```diff
@@ -0,0 +1,154 @@
+#!/bin/bash
+
+# This script register a bundle with the TRE API. It relies on the bundle
+# pre-existing in the remote repository (i.e. has been publish beforehand).
+
+set -o errexit
+set -o pipefail
+# set -o xtrace
+
+function usage() {
+    cat <<USAGE
+
+    Usage: $0 [-u --tre_url]  [-c --current] [-i --insecure]
+
+    Options:
+        -r, --acr-name        Azure Container Registry Name
+        -t, --bundle-type     Bundle type, workspace or workspace_service
+        -c, --current         Make this the currently deployed version of this template
+        -i, --insecure        Bypass SSL certificate checks
+        -u, --tre_url         URL for the TRE (required for automatic registration)
+        -a, --access-token    Azure access token to automatically post to the API (required for automatic registration)
+        -v, --verify          Verify registration with the API
+USAGE
+    exit 1
+}
+
+# if no arguments are provided, return usage function
+if [ $# -eq 0 ]; then
+    usage # run usage function
+fi
+
+current="false"
+verify=false
+
+while [ "$1" != "" ]; do
+    case $1 in
+    -u | --tre_url)
+        shift
+        tre_url=$1
+        ;;
+    -r | --acr-name)
+        shift
+        acr_name=$1
+        ;;
+    -t | --bundle-type)
+        shift
+        case $1 in
+        workspace)
+        ;;
+        workspace_service)
+        ;;
+        user_resource)
+        ;;
+        *)
+            echo "Bundle type must be workspace, workspace_service or user_resource, not $1"
+            exit 1
+        esac
+        bundle_type=$1
+        ;;
+    -c| --current)
+        current="true"
+        ;;
+    -i| --insecure)
+        insecure=1
+        ;;
+    -a | --access-token)
+        shift
+        access_token=$1
+        ;;
+    -v| --verify)
+        verify=true
+        ;;
+    *)
+        echo "Unexpected argument: '$1'"
+        usage
+        ;;
+    esac
+
+    if [[ -z "$2" ]]; then
+      # if no more args then stop processing
+      break
+    fi
+
+    shift # remove the current value for `$1` and use the next
+done
+
+explain_json=$(porter explain --reference ${acr_name}.azurecr.io/$(yq eval '.name' porter.yaml):v$(yq eval '.version' porter.yaml) -o json)
```

Comment:
is it worth having something like 
${acr_name:?} 
so that we check the variables are actually set?

Reply:
In other scripts we have checks that required variables are set. Adding that here would be good IMO - it feels easier to track down an error when the message is "--acr-name must be set" rather than a cryptic error from porter that the acr couldn't be found.

E.g. https://github.com/microsoft/AzureTRE/blob/main/devops/scripts/publish_register_bundle.sh#L75

Reply:
done.